server/controllers/list.js

```diff
@@ -605,70 +676,106 @@ const changeToMember = (req, resp) => {
         return resp.status(status).send({ message });
       }
 
-      resp.status(400).send({
-        message: 'List data not found'
-      });
+      resp.status(400).send({ message: 'List data not found' });
     });
 };
 
-const addMember = (req, resp) => {
+const removeMemberRole = (req, resp) => {
+  const { id: listId } = req.params;
+  const { userId } = req.body;
+  const {
+    user: { _id: ownerId }
+  } = req;
+
+  List.findOne({
+    _id: listId,
+    ownerIds: { $in: [ownerId] }
+  })
+    .populate('cohortId', 'memberIds ownerIds viewersIds')
+    .exec()
+    .then(doc => {
+      if (!doc) {
+        throw new BadRequestException("Can't remove member role");
+      }
+
+      const { memberIds, ownerIds } = doc;
+      const userIsMember = memberIds.some(id => id.equals(userId));
+
+      if (userIsMember) {
+        memberIds.splice(memberIds.indexOf(userId), 1);
+      }
+
+      const userIsOwner = ownerIds.some(id => id.equals(userId));
+
+      if (userIsOwner) {
+        ownerIds.splice(ownerIds.indexOf(userId), 1);
+      }
+
+      return doc.save();
+    })
+    .then(() =>
+      resp.status(200).send({
+        message: 'User has no member role'
+      })
+    )
+    .catch(err => {
+      if (err instanceof BadRequestException) {
+        const { status, message } = err;
+        return resp.status(status).send({ message });
+      }
+
+      resp.status(400).send({ message: 'List data not found' });
+    });
+};
+
+const addViewer = (req, resp) => {
   const {
     user: { _id: currentUserId }
   } = req;
   const { id: listId } = req.params;
   const { email } = req.body;
+  let list;
+  let user;
+  let cohortMembers;
 
-  List.findOne({ _id: listId, $or: [{ ownerIds: currentUserId }] })
-    .populate('cohortId', 'memberIds ownerIds')
+  List.findOne({
+    _id: listId,
+    $or: [{ ownerIds: currentUserId }]
+  })
+    .populate('cohortId', 'ownerIds memberIds')
     .exec()
     .then(doc => {
-      if (doc) {
-        const { cohortId: cohort, ownerIds } = doc;
+      list = doc;
 
-        return User.findOne({ email })
-          .exec()
-          .then(doc => {
-            const { _id, avatarUrl, displayName } = doc;
-            return { _id, avatarUrl, cohort, displayName, ownerIds };
-          })
-          .catch(() => {
-            throw new BadRequestException('User data not found.');
-          });
+      return User.findOne({ email }).exec();
+    })
+    .then(userData => {
```

Comment:
Your code is working good, but consider to add this check in this then block:
```
      if (!userData) {
        throw new BadRequestException(`There is no user of email: ${email}`);
      }
```
I think we should return information that there no user was found, no that some error occurred. This if statement will be a good place to send an invitation email in some future improvements.